lib/dynflow/scheduled_plan.rb

```diff
@@ -0,0 +1,59 @@
+module Dynflow
+  class ScheduledPlan < Serializable
+
+    include Algebrick::TypeCheck
+
+    attr_reader :execution_plan_uuid, :start_at, :start_before, :args
+
+    def initialize(world, execution_plan_uuid, start_at, start_before, args = [], args_serializer = nil)
+      @world               = Type! world, World
+      @execution_plan_uuid = Type! execution_plan_uuid, String
+      @start_at            = Type! start_at, Time
+      @start_before        = Type! start_before, Time, NilClass
+      @args                = Type! args, Array
+      @args_serializer     = Type! args_serializer, Serializers::Abstract
+    end
+
+    def execution_plan
+      @execution_plan ||= @world.persistence.load_execution_plan(@execution_plan_uuid)
+    end
+
+    def plan
+      execution_plan.root_plan_step.load_action
+      execution_plan.generate_action_id
+      execution_plan.generate_step_id
+      execution_plan.plan(*@args)
+    end
+
+    def timeout
+      execution_plan.root_plan_step.state = :error
+      execution_plan.root_plan_step.error = ::Dynflow::ExecutionPlan::Steps::Error.new("Execution plan could not be started before set time (#{@start_before})")
+      execution_plan.root_plan_step.save
+      execution_plan.execution_history.add 'timeout', @world.id
+      execution_plan.update_state :stopped
+    end
+
+    def execute
+      @world.execute(execution_plan.id)
+    end
+
+    def to_hash
+       recursive_to_hash :execution_plan_uuid => @execution_plan_uuid,
+                         :start_at            => time_to_str(@start_at),
+                         :start_before        => time_to_str(@start_before),
+                         :args                => @args_serializer.serialize,
+                         :args_serializer     => @args_serializer.class.name
+    end
+
+    # @api private
+    def self.new_from_hash(world, hash, *args)
+      serializer = hash[:args_serializer].constantize.new
```

Comment:
What happens, when the serializer class is removed in the mean time: think of a case, when somebody would rename the serializer in new version of the software, and one updates the application, while something scheduled for next week: Just makring the plan as failed would work here.